### ISSUE TITLE
Does slick-pg support individual ARRAY element access?

### ISSUE THREAD

**Issue by iceberg901** [tminglei/slick-pg#157]
I am trying to write a query with a condition that would look like this in SQL:

```sql
WHERE t1.id = t2.linked_ids[1]
```

Is there some way to do this with slick-pg without using plain SQL?

I can't find any way to do the direct subscript access with slick-pg.  So for example let's say the linked_ids column is defined as

```scala
val linkedIds: Column[List[Long]] = column[List[Long]]("linked_ids", O.NotNull, O.Default(List()), O.DBType("BIGINT[]"))
```

the one cannot use subscript access in scala like:

```scala
linkedIds[1]
```

**Comment by tminglei** [comment]
No, there isn't. And I didn't find a way to implement it yet.

**Comment by iceberg901** [comment]
Thanks as always for the quick response!

**Comment by mdedetrich** [comment]
I am also wondering how you would implement this, I don't think individual "index" access for columns is possible (yet) in how slick is set up

**Comment by tminglei** [comment]
@mdedetrich, currently, I can't find a way to generate `WHERE t1.id = t2.linked_ids[1]` by using `slick`.

**Comment by mdedetrich** [comment]
Yeah thats what I meant, I don't think its physically possible, might have to make an issue at https://github.com/slick/slick for it?

**Comment by tminglei** [comment]
@mdedetrich  yes, let's file an issue to slick.

**Comment by alfonsovng** [comment]
I suppose that this is not available yet ;(.

**Comment by coreycaplan3** [comment]
One thing you could do as a solid workaround is make a static function that composes the access. For instance, I wrote the following in Pg SQL:
```
create or replace function firstElement(any_array anyarray)
  returns anyelement
  immutable
  language plpgsql
as
$$
  begin
    return any_array[1];
  end;
$$;
```

Then in scala:
```
def firstElementFn[T: TypedType](array: Rep[List[T]]): Rep[Option[T]] = {
    SimpleFunction.unary[List[T], Option[T]]("firstElement").apply(array)
}
```

Usage:
```
metricTick.map(t => firstElementFn(arrayAgg(t.openPrice.?).sortBy(t.metricTimestamp.asc)))
```
where metricTick is a `Query` being used in a `Group By` and `t` represents the underlying `table`.
